### PR TITLE
Dynamically configure the line detector

### PR DESCRIPTION
**Pull request by aftersomemath**

### CODE REVIEW

File: cfg/LineDetector.cfg [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env python
+PACKAGE = "iarc7_vision"
+
+from dynamic_reconfigure.parameter_generator_catkin import *
+
+gen = ParameterGenerator()
+
+gen.add("pixels_per_meter",    int_t,    0, "Pixels per meter for image scale", 250,  0, 1000)
```
Comment (amiller27) [inline cfg/LineDetector.cfg]: Can you get rid of the weird spacing here?

File: src/VisionNode.cpp [hunk 2]
```diff
@@ -11,43 +11,78 @@
...
+#include <dynamic_reconfigure/server.h>
+#include <iarc7_vision/LineDetectorConfig.h>
+
 #include <sensor_msgs/image_encodings.h>
 
 #include "iarc7_vision/GridLineEstimator.hpp"
 
-void getLineExtractorSettings(const ros::NodeHandle& private_nh,
+void getLineExtractorSettings(iarc7_vision::LineDetectorConfig &config,
+                              uint32_t,
```
Comment (amiller27) [inline src/VisionNode.cpp]: We don't need this

File: src/VisionNode.cpp [hunk 3]
```diff
@@ -119,8 +154,24 @@ int main(int argc, char **argv)
...
+    boost::function<void(iarc7_vision::LineDetectorConfig &config,
+                         uint32_t level)> line_extractor_lambda =
+        [&](iarc7_vision::LineDetectorConfig &config, uint32_t level) {
+            getLineExtractorSettings(config,
+                                     level,
+                                     private_nh,
+                                     line_extractor_settings);
+        };
+
+    settings_callback = boost::bind(line_extractor_lambda, _1, _2);
```
Comment (amiller27) [inline src/VisionNode.cpp]: Don't need this

File: cfg/LineDetector.cfg (hunk 1 above)
Comment (aftersomemath) [reply cfg/LineDetector.cfg]: lol

File: src/VisionNode.cpp (hunk 2 above)
Comment (aftersomemath) [reply src/VisionNode.cpp]: Fixed

File: src/VisionNode.cpp (hunk 3 above)
Comment (aftersomemath) [reply src/VisionNode.cpp]: Fixed

**Review by amiller27 — APPROVED** [review]

**Pull request merged**